### DISCUSSION TITLE
Dual Wielding for Flare Version 1.13?

### DISCUSSION THREAD

**Discussion by WithinAmesia** [flareteam/flare-game#852] [Ideas]
So I achieved victory for my Dual Wielding goals in the Flare Engine but I figured it better if I also helped the community as well with successful Flare Game Dual Wielding: https://opengameart.org/content/10-off-hand-weapon-animations-for-both-the-heroine-and-hero-for-the-flare-rpg-engine
Should I make a bunch of Flare Game off-hand weapons that are ~mirrors of the main hand versions? If so I would intend to have dual wielding being equal to two-handed weapon's damage output with main-hand + off-hand of equal tier. Now that we have the dual wielding animations; would the little bit of extra work to bring Dual Wielding support into the Flare Game be desired as a feature for the upcoming Flare Version 1.13?

I figured since I targeted the 10 base one-hand melee weapons in Flare that I might as well make versions that can go into the Flare Game. In that I feel should do something like this as a thank you to all the people who helped myself become better creatively. Also by adding more content to Flare for the community to enjoy it would in a way go full circle fulfilling the spirit of giving back to the community; especially after I received so much help from the community. Also it would be cool to see my favourite play-style be alive and well in the Flare Game:-).

I am wondering how to format / share these new off-hand weapons? Although I want to ask first if this Flare Game Dual Wielding work idea is desired before taking the time of making and sharing a (somehow, a .zip?) worked on base 1.12 Flare Game. As in should I work towards adding in all of the off-hand melee weapons to mirror the main handed ones in order to bring Dual Wielding support the upcoming Flare Version 1.13? If so what is the easiest format / way to share these new off-hand / dual wielding items like this to easily share; if such work is desired?

**Comment by dorkster** [comment]
Dual wielding is not something I'm looking to add to flare-game from a game design perspective. I'm happy with the existing weapon balance, and I think adding a bunch of off-hand weapons would needlessly complicate things.

While the work you've done is a neat proof-of-concept, I think it would fall short of players' expectations. In my mind, players would expect to be able to take any regular weapon (or shield for that matter), and put them in either hand. So in other words, there wouldn't even *be* a distinction between "main hand" and "off hand". Instead, the slots could each just be named "Weapon / Shield".

To do this properly, some work would need to happen on the engine side. We would need to create a some sort of divide between equipment slots in the inventory and the player's sprite layers. It would need to be robust enough so that no matter which equip slots the items are in, the correct sprite layers would be used. Item definitions would need to support multiple animations, and each animation would need to define which sprite layer it targets. As an example, let's say all the melee/mental weapons have animations for either hand, but shields and bows only have "off hand" animations:

- A bow placed in either hand would only have a single animation (the current off hand animation).
- The player replaces the bow with a melee weapon in the "off hand" slot. There would be no layer conflicts, so we could use the off hand animation for that weapon.
- The player then equips a shield in the "main hand" slot. The shield would take the off hand layer because it only has an off hand animation. So the melee weapon, despite wanting to show on the off hand due to its slot, would have to fall back to the main hand.
- Finally, the player replaces the shield with another melee weapon. Each slot would map directly to its matching sprite layer, so no layer conflict.

**Reply by WithinAmesia** [reply]
So like in most bigger scale / 3D / newer RPGs with dual wielding a 'One Handed' class / category of item / weapon (and the supporting mechanics) would be needed in order to bring Flare Dual Wielding in line with current common player expectations?

As in where if a one-handed ~weapon was placed in the Main Hand slot it would use main hand animations and if it was put into Off-Hand slot it would support off hand animations? Given how Flare handles the male / female animations for items where one item can support two animation sets; would a similar style of solution be plausible for one-handed ~weapons being put into Main Hand for main hand / right hand animations and Off Hand for off hand / left hand animations?

Also if a ~weapon only has one set of animations (main hand or off hand only) it slot restricts it self (via an item type distinct from One Hand such as Main Hand or Off Hand or even Two Hand) to be a main hand / two-handed only (like a main hand only ~bastard sword / bows and two-handed weapons) or off hand only (like the sling shot / ~off-hand frills). Would this style of implementation make sense and bring the mechanical aspect of Flare Dual Wielding in line with common player expectations in how dual wielding works in most larger scale / newer RPGs (World of Warcraft, Diablo III, Path of Exile? etc.)

**Reply by dorkster** [reply]
Yes, that all sounds about right. Any weapon in either hand, and let the engine determine which animations to use based on what's available. That's the theory at least. Not sure how well it would work in practice.

**Reply by WithinAmnesia** [reply]
> 
> 
> Yes, that all sounds about right. Any weapon in either hand, and let the engine determine which animations to use based on what's available. That's the theory at least. Not sure how well it would work in practice.

Only one way to find out :-P. Iterate, iterate, iterate:-).
